src/error.rs

```diff
@@ -325,18 +325,21 @@ impl<I: fmt::Display + Hash + Eq, S: Span> fmt::Display for Simple<I, S> {
                     None => format!("end of input"),
                 },
             )?,
-            _ => write!(
-                f,
-                " but one of {} was expected",
-                self.expected
-                    .iter()
-                    .map(|expected| match expected {
-                        Some(x) => format!("'{}'", x),
-                        None => format!("end of input"),
-                    })
-                    .collect::<Vec<_>>()
-                    .join(", ")
-            )?,
+            _ => {
+                write!(
+                    f,
+                    "found {:?}, but one of {:?} was expected",
+                    found,
+                    self.expected
+                        .iter()
+                        .map(|expected| match expected {
+                            Some(x) => format!("{}", x),
+                            None => found.to_string(),
```

Comment:
I'm wondering, why use `found` here?

Reply:
I'm not very familiar with the codebase and I suppose I tried to make as small a change as possible. I don't like this, either. I don't have a good explanation for this.

With that said, I will clean this section up. Thank you for pointing this out.